kmultiversx/src/kmultiversx/kdist/mx-semantics/elrond-config.md

```diff
@@ -615,72 +646,58 @@ TODO: Implement [reserved keys and read-only runtimes](https://github.com/Elrond
                 ~> resetCallstate
                 ~> transferFunds(FROM, TO, VALUE)
                 ~> transferESDTs(FROM, TO, ESDT)
-                ~> newWasmInstance(TO, CODE)
-                ~> mkCall(TO, FUNCNAME, VMINPUT)
+                ~> callContractAux(FROM, TO, FUNCNAME, VMINPUT)
                 ~> #endWasm
                    ...
         </commands>
+        <vmOutput> _ => .VMOutput </vmOutput>
+        <logging> S => S +String " -- callContract " +String #parseWasmString(FUNCNAME) </logging>
+
+    rule [callContractAux]:
+        <commands> callContractAux(_FROM, TO, FUNCNAME, VMINPUT)
+                => newWasmInstance(TO, CODE)
+                ~> mkCall(TO, FUNCNAME, VMINPUT)
+                   ...
+        </commands>
         <account>
           <address> TO </address>
           <code> CODE </code>
           ...
         </account>
-        <vmOutput> _ => .VMOutput </vmOutput>
-        <logging> S => S +String " -- callContract " +String #parseWasmString(FUNCNAME) </logging>
-        <instrs> (#waitCommands ~> _) #Or .K </instrs>
+        <instrs> .K </instrs>
```

Comment:
Just making sure: when calling callContractAux, the instrs cell is always empty, right?

Reply:
Yes, `<instrs>` is emptied in `callContract => resetCallState`.